File: panaderia-backend/src/main/java/com/besoft/panaderia/service/UbigeoService.java
package com.besoft.panaderia.service;

import java.util.List;

import org.springframework.web.bind.annotation.RequestBody;

import com.besoft.panaderia.dto.request.DepartamentoRequest;
import com.besoft.panaderia.dto.request.DistritoRequest;
import com.besoft.panaderia.dto.request.ProvinciaRequest;
import com.besoft.panaderia.dto.response.DepartamentoResponse;
import com.besoft.panaderia.dto.response.DistritoResponse;
import com.besoft.panaderia.dto.response.OutResponse;
import com.besoft.panaderia.dto.response.PaisResponse;
import com.besoft.panaderia.dto.response.ProvinciaResponse;

public interface UbigeoService {

	public OutResponse<List<PaisResponse>> listarPais();

	public OutResponse<List<DepartamentoResponse>> listarDepartamento(@RequestBody DepartamentoRequest req);

	public OutResponse<List<ProvinciaResponse>> listarProvincia(@RequestBody ProvinciaRequest req);

	public OutResponse<List<DistritoResponse>> listarDistrito(@RequestBody DistritoRequest req);
}

File: panaderia-frontend/src/main/panaderia-frontend/src/app/modules/intranet/services/ubigeo.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { DepartamentoRequest } from '../dto/request/departamento.request';
import { DistritoRequest } from '../dto/request/distrito.request';
import { ProvinciaRequest } from '../dto/request/provincia.request';
import { DepartamentoResponse } from '../dto/response/departamento.response';
import { DistritoResponse } from '../dto/response/distrito.response';
import { OutResponse } from '../dto/response/out.response';
import { PaisResponse } from '../dto/response/pais.response';
import { ProvinciaResponse } from '../dto/response/provincia.response';

@Injectable()
export class UbigeoService {

  constructor(private http: HttpClient) { }

  public listarPais(): Observable<OutResponse<PaisResponse[]>> {
    return this.http.post<OutResponse<PaisResponse[]>>(`${environment.WsPanaderiaBackend}/ubigeo/listarPais`, {});
  }

  public listarDepartamento(req: DepartamentoRequest): Observable<OutResponse<DepartamentoResponse[]>> {
    return this.http.post<OutResponse<DepartamentoResponse[]>>(`${environment.WsPanaderiaBackend}/ubigeo/listarDepartamento`, req);
  }

  public listarProvincia(req: ProvinciaRequest): Observable<OutResponse<ProvinciaResponse[]>> {
    return this.http.post<OutResponse<ProvinciaResponse[]>>(`${environment.WsPanaderiaBackend}/ubigeo/listarProvincia`, req);
  }

  public listarDistrito(req: DistritoRequest): Observable<OutResponse<DistritoResponse[]>> {
    return this.http.post<OutResponse<DistritoResponse[]>>(`${environment.WsPanaderiaBackend}/ubigeo/listarDistrito`, req);
  }
}
